### ISSUE TITLE
China Hacker(s) stop hacking inside Internet Center when pressing STOP key

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#767]
China Hacker(s) stop hacking inside Internet Center when pressing STOP key. Hacker(s) will have to be evacuated and reentered to start hacking again.

**Comment by xezon** [comment]
Enlima29:
> I once meddled with hacker van, where those are mobile internet center where hackers can enter them to generate money. I accidentally pressed stop command on it and the hackers inside stopped hacking. But then if I disable it by EMP or subdual, when the van is back online and the hackers inside starts hacking again for some reason. So I guess make a special button for the internet center that will disable (stun) it for like 0.1 second and have the button have S as hotkey so it will be triggered along with the stop command?

**Comment by DoMiNaToRuk2** [comment]
Please stop that stop

**Comment by xezon** [comment]
Unfortunately, the Sat Hack upgrade button is bound to S. So pressing S to get or use upgrade will also stop the Hackers.

![shot_20230215_185826_1](https://user-images.githubusercontent.com/4720891/219114020-9e65e5d8-6f0b-43eb-9b7c-856145969681.jpg)

**Comment by xezon** [comment]
Removing S shortcut from button does not prevent bug.

**Comment by xezon** [comment]
```
  Behavior InternetHackContain
    ContainMax
    EnterSound
    ExitSound
    DamagePercentToUnits
    BurnedDeathToUnits
    AllowInsideKindOf
    ForbidInsideKindOf
    PassengersAllowedToFire
    PassengersInTurret
    NumberOfExitPaths
    DoorOpenTime
    WeaponBonusPassedToPassengers
    AllowAlliesInside
    AllowEnemiesInside
    AllowNeutralInside
    DeathTypes
    VeterancyLevels
    ExemptStatus
    RequiredStatus
    Slots
    ScatterNearbyOnExit
    OrientLikeContainerOnExit
    KeepContainerVelocityOnExit
    GoAggressiveOnExit
    ResetMoodCheckTimeOnExit
    DestroyRidersWhoAreNotFreeToExit
    ExitBone
    ExitPitchRate
    InitialPayload
    HealthRegen%PerSec
    ExitDelay
    ArmedRidersUpgradeMyWeaponSet
    DelayExitInAir
```

**Comment by xezon** [comment]
Using `ContainMax` instead of `Slots` does not work.

**Comment by xezon** [comment]
Adding `PassengersInTurret = Yes` does not fix it.

**Comment by xezon** [comment]
Looks like code issue. The behaviour is identical on other structures. For example loading rocket men into Bunker, shooting and then pressing stop on Bunker will make the rocket men stop firing. So we probably would need to add a new feature like `ForwardUnitCommandsToPassengers = No` with the default being `Yes`.

**Comment by commy2** [comment]
Or a flag that disabled Stop entirely. It can then be given to Scud Storm as well.

You still need to pass the evacuate command.

**Comment by xezon** [comment]
Evacuate command is structure button.

**Comment by MTKing4** [comment]
Then put the sathack on a different hotkey? Maybe K
Worst case scenario for sathack is to add stop button

And maybe start hacking button as well if that's possible

**Comment by xezon** [comment]
Yes as long as we cannot fix this issue, remapping hack to different hotkey is desired. We could map it to H for "Hack".

**Comment by commy2** [comment]
H is "Home", brings you to your oldest CC.  Ctrl+H is hero, brings you to your oldest hero.

**Comment by xezon** [comment]
F

**Comment by MTKing4** [comment]
Yeah i thought about H but as commy Said it's not available, closest we have is `K` or `T`

I wouldn't put any of the following `A` `I` `L` `C` `E` as they're too common in most words

**Comment by commy2** [comment]
Why not A? It has no attack move. A is a good key for buildings, and it's easy to reach (unlike, e.g. L)

**Comment by MTKing4** [comment]
I don't like it, it's common vowel, and also known for attack move

**Comment by MTKing4** [comment]
If you like close letters then T is pretty close

**Comment by xezon** [comment]
A looks good to me. Close to former S.

**Comment by MTKing4** [comment]
I'm going to stop my hackers a lot every time i issue an attack move while having my IC selected

**Comment by xezon** [comment]
Perhaps a silly thought but I wonder could the ability to stop the hackers perhaps be useful in some cases? Since the enemy can see the green numbers of the active Hackers above the Internet Center, one could try to momentarily conceal their presence and make it look as if the Internet Center is empty, for example when the enemy USA player activates their Strategy Center Battle Plan and uses the Satellite Intelligence ability to see the whole map.

**Comment by ImTimK** [comment]
> I'm going to stop my hackers a lot every time i issue an attack move while having my IC selected
> 
> > if the first letter of the ~Unit/Structure/Upgrade~ _**Command**_ name isn't available, the next letter is used.
> 
> s**T**op

How so, why would you press S when it's on A?

s**A**tellite hack

You don't use any A's, S's, G's or X's (attack move, stop, guard mode, scatter) in pruduction facilities? Your hotkey options sound very limited and decentralized if you get confused that easy.
